File: 111-minimum-depth-of-binary-tree/minimum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def minDepth(self, root):
        """
        :type root: Optional[TreeNode]
        :rtype: int
        """
        q=deque([root])
        height=1
        
        if root is None:
            return 0
        
        while q:
            for _ in range(len(q)):
                node=q.popleft()
                if node.left is None and node.right is None:
                    return height
                if node.left:
                    q.append(node.left)
                if node.right:
                    q.append(node.right)
            height+=1 

File: 111-minimum-depth-of-binary-tree/minimum-depth-of-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int minDepth(TreeNode* root) {
        if(root==nullptr) return 0;

        deque<tuple<TreeNode*,int>> bfs;
        bfs.push_back({root,1});
        
        while(!bfs.empty()){
            auto [node, depth] = bfs.front();
            bfs.pop_front();

            if(node->left==nullptr && node->right==nullptr) return depth;
            if(node->left!=nullptr) bfs.push_back({node->left,depth+1});
            if(node->right!=nullptr) bfs.push_back({node->right,depth+1});
        }
        return -1;
    }
};